### ISSUE TITLE
Windows10 pytorch NO Final result

### ISSUE THREAD

**Issue by cjissmart** [microsoft/nni#1752]
**Short summary about the issue/question**:
When I use pytorch, I can't get result and some unusual thing happened
**Brief what process you are following**: 
When I run the example: mnist-pytorch, I can't get the trial's final result. And the intermediate result noly updates once, which means only one intermediate result during one trial.The same code is run well on ubuntu. Besides, it's ok when I use tensorflow.
The figures 

**How to reproduce it**: 
I don't know how to reproduce it, but the problem has been found by #1558 .
Unfortunately, #1558 didn't show the solution.

Each trial has only one intermediate result
![image](https://user-images.githubusercontent.com/37531166/69208819-0082f500-0b90-11ea-946f-7f70423750cc.png)
Don't exist final results for all trials
![image](https://user-images.githubusercontent.com/37531166/69208926-5061bc00-0b90-11ea-84ea-90f294403d51.png)

Notation: if I use tensorflow, everything is ok !
**nni Environment**:
- nni version:  v1.1
- nni mode(local|pai|remote): local
- OS: windows10
- python version: 3.7
- is conda or virtualenv used?: conda virtualenv 
- is running in docker?: no

**need to update document(yes/no)**:

**Anything else we need to know**:
It seems thath trial's trial.log only save pytorch's print. For python print function, there is Null. And the file is rewrite when a epoch finishes and new epoch starts.
![image](https://user-images.githubusercontent.com/37531166/69209056-ccf49a80-0b90-11ea-9c50-5555da062eca.png)

**Comment by ultmaster** [comment]
Hi. Thanks for reaching out.

I tried running mnist-pytorch with the exact same settings as you on Windows, and everything is fine.

![image](https://user-images.githubusercontent.com/8463288/69212637-d683ff80-0b9c-11ea-8cf2-bf650dfb5911.png)

The issue with `trial.log` really intrigues me. Have you tried running `mnist.py` alone? Does it show the same result?

Meanwhile, try to find `.nni/metrics` in your nni output directory (the same directory where you find your `trial.log`), and paste the content here.

**Comment by cjissmart** [comment]
Appreciate!!!
This is the metrics. It's so strange! As the figure show, the whole file exist so many null, and all useful message is the right side of the figure.
![image](https://user-images.githubusercontent.com/37531166/69224377-633bb700-0bb7-11ea-84b7-a5b7a6483caa.png)

**Comment by ultmaster** [comment]
@cjissmart Sorry I'm not a fan of Notepad++. I'm not sure `NUL` means corrupted or actually zero. Could you open your file in binary mode and show the data presented?

It would be better if you can show several versions of this file during the run of your experiment.

**Comment by cjissmart** [comment]
![image](https://user-images.githubusercontent.com/37531166/69244946-e9202800-0be0-11ea-83cb-a9da889281ba.png)
I dont know this figure is right . In binary mode, `20` indicates `NUL`
And other versions are same as the figure. The difference between different metrics is the end characters of files.

**Comment by cjissmart** [comment]
![image](https://user-images.githubusercontent.com/37531166/69244946-e9202800-0be0-11ea-83cb-a9da889281ba.png)
I dont know this figure is right . In binary mode, `20` indicates `NUL`
And other versions are same as the figure. The difference between different metrics is the end characters of files.

**Comment by cjissmart** [comment]
I'm sure that the same code is run well on other windows' computer.
So strange!!!

**Comment by ultmaster** [comment]
@SparkSnail Any clue on why this happen on Windows?

**Comment by SparkSnail** [comment]
Hi @cjissmart , could you please try using `http://localhost:8080/api/v1/nni/metric-data` to get the metrics data and paste them here after you start an experiment in your Windows machine?

**Comment by cjissmart** [comment]
I'm sorry for responsing not in time.
![image](https://user-images.githubusercontent.com/37531166/69479563-9ee5b380-0e39-11ea-9c6a-ba0cc621b1ff.png)
this is what `http://localhost:8080/api/v1/nni/metric-data` shows.
![image](https://user-images.githubusercontent.com/37531166/69479582-d18fac00-0e39-11ea-9144-fe7c6c69648b.png)
this is what `http://localhost:8080/api/v1/nni/trial-jobs` shows.
The fisrt two trials have finished, and the last one is running.
Can it be helpful?

**Comment by cjissmart** [comment]
The Same code is run well if I don't use GPU.
When I use CUDA_VISIBLE_DEVICES to select GPU and set gpu_num being 0, CUDA_VISIBLE_DEVICES commad is unavailable. You know how to fix it?

**Comment by SparkSnail** [comment]
Hi @cjissmart , do you mean that if you do not use GPU in the experiment, you could get the `final_result` of trials? If you want to select gpu in NNI's trial job, you could use `gpu_indices` field in config file, [refer](https://nni.readthedocs.io/en/latest/Tutorial/ExperimentConfig.html#configuration-spec).

**Comment by cjissmart** [comment]
I modify mnist-pytorch's config.yml as the figure shows:
![image](https://user-images.githubusercontent.com/37531166/69626781-4fdf8e80-1084-11ea-907f-27387022bad1.png)
However, it doesn't fix the bug.
It seems that it always happens when use pytorch with gpu.

**Comment by SparkSnail** [comment]
If you set `gpuIndices` value, you need to set `gpuNum` a non-zero number too, or it would not work. Does nni works on a tensorflow example in your machine?

**Comment by scarlett2018** [comment]
> If you set `gpuIndices` value, you need to set `gpuNum` a non-zero number too, or it would not work. Does nni works on a tensorflow example in your machine?

@cjissmart any update for this thread?

**Comment by liubola** [comment]
same question
![image](https://user-images.githubusercontent.com/22904295/79310200-1ffb2f80-7f2e-11ea-8b72-177ef49bde1d.png)

**Comment by scarlett2018** [comment]
> > If you set `gpuIndices` value, you need to set `gpuNum` a non-zero number too, or it would not work. Does nni works on a tensorflow example in your machine?
> 
@liubola - had you try what @SparkSnail suggest above? any feedback?

**Comment by liubola** [comment]
this is my config, and it is not worked.
mnist-pytorch
![image](https://user-images.githubusercontent.com/22904295/79310480-9e57d180-7f2e-11ea-81cd-4b1642e0788b.png)

**Comment by liubola** [comment]
> > > 如果设置`gpuIndices`值，则也需要设置`gpuNum`一个非零的数字，否则它将不起作用。nni是否可以在您机器上的tensorflow示例上工作？
> 
> @liubola-您尝试过@SparkSnail以上建议吗？任何反馈？

this is my config, and it is not worked.
mnist-pytorch
![image](https://user-images.githubusercontent.com/22904295/79310480-9e57d180-7f2e-11ea-81cd-4b1642e0788b.png)

**Comment by liubola** [comment]
> > > If you set `gpuIndices` value, you need to set `gpuNum` a non-zero number too, or it would not work. Does nni works on a tensorflow example in your machine?
> 
> @liubola - had you try what @SparkSnail suggest above? any feedback?

It is worked if i just run "python mnist.py" in the terminal, but faild through nni

**Comment by SparkSnail** [comment]
Hi @liubola, could you please use `nnictl create --debug` mode to run an NNI experiment in your machine, and show me the content of nniManager.log and `.metric` file? The `.metric` file can be found under trial's log folder.

**Comment by liubola** [comment]
[nnimanager.log](https://github.com/microsoft/nni/files/4501776/nnimanager.log)

I found metric file under .nni folder.
ME000094{"parameter_id": 0, "trial_job_id": "ljspa", "type": "FINAL", "sequence": 0, "value": "99.3"}

**Comment by SparkSnail** [comment]
Hi @liubola, this issue is probably a potential bug of NNI, I've add more information and did some hotfix, could you please use following command to install NNI, and provide your nniManager.log?
```
git clone -b dev-debug https://github.com/SparkSnail/nni.git
cd nni
powershell -ExecutionPolicy Bypass -file install.ps1
```

**Comment by liubola** [comment]
[nnimanager (2).log](https://github.com/microsoft/nni/files/4549826/nnimanager.2.log)
@SparkSnail

**Comment by SparkSnail** [comment]
> [nnimanager (2).log](https://github.com/microsoft/nni/files/4549826/nnimanager.2.log)
> @SparkSnail

Does this experiment still has no `final_result` in WebUI? If you use `nnictl log stdout` command, is there any information?

**Comment by LuckGuySam** [comment]
I have the same problem！！
Anyone has the solution ?
My environment : 
  pythorch version: 1.5.0
  nni version: v1.9
  nni mode(local|pai|remote): local
  OS: windows10
  python version: 3.7
  is conda or virtualenv used?: conda virtualenv
  is running in docker?: no
problem : 
  It will successful on cpu, but if work on gpu then the metric can't show on web. The output metric  is same @cjissmart !

**Comment by QuanluZhang** [comment]
@LuckGuySam thanks for reporting this issue. could you help us debug this issue? we can discuss on gitter (https://gitter.im/Microsoft/nni?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge).

**Comment by scarlett2018** [comment]
@LuckGuySam @cjissmart @QuanluZhang @SparkSnail - Has the problem been solved？

**Comment by kvartet** [comment]
@kvartet - need to test whether this issue exists in the latest version.

**Comment by scarlett2018** [comment]
> @LuckGuySam @cjissmart @QuanluZhang @SparkSnail - Has the problem been solved？

Closing as overdued.
